lib/ASTGen/Sources/ASTGen/Literals.swift

```diff
@@ -11,18 +11,200 @@
 //===----------------------------------------------------------------------===//
 
 import ASTBridging
+@_spi(Compiler) import SwiftParser
 import SwiftSyntax
 
 extension ASTGenVisitor {
-  func generate(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedStringLiteralExpr {
-    let openDelimiterOrQuoteLoc = self.generateSourceLoc(node.openingPounds ?? node.openingQuote)
+  func generate(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedExpr {
+    if node.segments.allSatisfy({ $0.is(StringSegmentSyntax.self) }) {
+      return self.generateStringLiteralExpr(stringLiteralExpr: node).asExpr
+    } else {
+      return self.generateInterpolatedStringLiteralExpr(stringLiteralExpr: node).asExpr
+    }
+  }
 
-    // FIXME: Handle interpolated strings.
-    // FIXME: Avoid 'String' instantiation
-    var segment = node.segments.first!.as(StringSegmentSyntax.self)!.content.text
-    return segment.withBridgedString { bridgedSegment in
-      return .createParsed(self.ctx, value: bridgedSegment, loc: openDelimiterOrQuoteLoc)
+  func generateStringLiteralExpr(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedStringLiteralExpr {
+    let stringLiteralKind = node.stringLiteralKind ?? .singleLine
+    let delimiterLength = node.delimiterLength
+
+    var value: String = ""
+    for case .stringSegment(let seg) in node.segments {
+      seg.appendUnescapedLiteralValue(
+        stringLiteralKind: stringLiteralKind,
+        delimiterLength: delimiterLength,
+        to: &value
+      )
+    }
+    return value.withBridgedString {
+      BridgedStringLiteralExpr.createParsed(
+        self.ctx,
+        value: $0,
+        loc: self.generateSourceLoc(node)
+      )
+    }
+  }
+
+  func generateInterpolatedStringLiteralExpr(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedInterpolatedStringLiteralExpr {
+    let stringLiteralKind = node.stringLiteralKind ?? .singleLine
+    let delimiterLength = node.delimiterLength
+    let startLoc = self.generateSourceLoc(node)
+    let afterQuoteLoc: BridgedSourceLoc = {
+      var l = startLoc
+      if let pound = node.openingPounds {
+        l = l.advanced(by: pound.trimmedLength.utf8Length)
+      }
+      l = l.advanced(by: node.openingQuote.trimmedLength.utf8Length)
```

Comment:
This wouldn’t handle any trivia in between the pound and the opening quote. There shouldn’t be any, in theory but you never know how the parser might change. So, I think would be good to just add that as well.